model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro.py

```diff
@@ -1109,36 +1090,19 @@ def run_predictor_step(
             iau_wgt_dyn=self._config.iau_wgt_dyn,
             is_iau_active=self._config.is_iau_active,
             limited_area=self._grid.limited_area,
-            iadv_rhotheta=self._config.iadv_rhotheta,
-            igradp_method=self._config.igradp_method,
             nflatlev=self._vertical_params.nflatlev,
             nflat_gradp=self._vertical_params.nflat_gradp,
-            start_edge_halo_level_2=self._start_edge_halo_level_2,
-            end_edge_halo_level_2=self._end_edge_halo_level_2,
             start_edge_lateral_boundary=self._start_edge_lateral_boundary,
-            end_edge_halo=self._end_edge_halo,
             start_edge_lateral_boundary_level_7=self._start_edge_lateral_boundary_level_7,
             start_edge_nudging_level_2=self._start_edge_nudging_level_2,
-            end_edge_local=self._end_edge_local,
-            end_edge_end=self._end_edge_end,
+            end_edge_nudging=self._end_edge_nudging,
             horizontal_start=gtx.int32(0),
```

Comment:
should we use a `0` encoded as starting from `LATERAL_BOUNDARY`.

Reply:
maybe I'll keep the `0` for now and we target improvements in this area in a more coherent way later.